### PR TITLE
{lib}[GCCcore/9.3.0] Mesa v20.0.2, zstd v1.4.4, lz4 v1.9.2

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

Main changes with the update of Mesa to v20.0.2:
* This PR follows the proposed changes for Mesa in #9764 and https://github.com/easybuilders/easybuild-easyblocks/pull/1892 , which automatically sets appropriate `configopts` for the system's architecture. The `mesa` easyblock seems very close to be ready. If it turns out that it is too much work to have it merged on time for `2020a`, this easyconfig for Mesa can be switched back to generic `mesonninja` very easily.
* As a consequence of using per arch `configopts`, patch `Mesa-19.2.1-meson-avoid-march.patch` is no longer necessary.
* Comments in the header of the easyconfig updated reflecting all these changes
* Added new dependency on `zstd`
* Removed dependency on `nettle`, which is no longer required
* Depends on `LLVM` v9.0.1 (#10133). For the time being this seems the candidate to be merged into `2020a`.
* `sanity_check_paths` has moved to the `mesa` easyblock:
    * ~~Libraries in `sanity_check_paths` have been reduced to those that I think are not arch dependent (i.e. ignoring libs such as `libswrAVX2.so`). Let me know if those are not working on your system.~~
    * Headers in `sanity_check_paths` have also changed quite a lot because many of them are no longer installed in builds based on `libglvnd` (check `include/meson.build` in the build directory).

Output of `glxinfo | grep OpenGL` in my test system is as good as it was with Mesa v19 and `glxgears` shows similar performance
```
OpenGL vendor string: VMware, Inc.
OpenGL renderer string: llvmpipe (LLVM 9.0.1, 256 bits)
OpenGL core profile version string: 3.3 (Core Profile) Mesa 20.0.2
OpenGL core profile shading language version string: 3.30
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.1 Mesa 20.0.2
OpenGL shading language version string: 1.40
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 20.0.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
```

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/1244691746bde37ccc65f131d9cbd982 for a full test report.

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/ab03a75ca30f5ecf992541eb23cf8440 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 8 out of 8 (3 easyconfigs in this PR)
b-an04.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/a88d2b52f986d7193b7e1a615ad3375b for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 25 out of 25 (3 easyconfigs in this PR)
bear-pg0211u03a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz, Python 2.7.5
See https://gist.github.com/f3b021228381cfc52d74784cbf0b791c for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 25 out of 25 (3 easyconfigs in this PR)
bear-pg0305u03a.bear.cluster - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
See https://gist.github.com/a14d7d3df737bbe2480018dbfb8493c8 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 25 out of 25 (3 easyconfigs in this PR)
bber0501u03a.bb2.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/01f79d815d45d10c14b1278d49bb24ce for a full test report.

File: easybuild/easyconfigs/m/Mesa/Mesa-20.0.2-GCCcore-9.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+# symlink indirect to mesa GLX, similar to Debian, see
+# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=881789
+# This helps in certain X forwarding situations (e.g. XQuartz)
+postinstallcmds = ["ln -s libGLX_mesa.so.0 %(installdir)s/lib/libGLX_indirect.so.0"]
+
+# Tells libglvnd where to find EGL libraries
+modextrapaths = {"__EGL_VENDOR_LIBRARY_DIRS": "share/glvnd/egl_vendor.d"}
+
+sanity_check_paths = {
```
Comment (boegel) [inline easybuild/easyconfigs/m/Mesa/Mesa-20.0.2-GCCcore-9.3.0.eb]: if https://github.com/easybuilders/easybuild-easyblocks/pull/1892 is enhanced with the changes proposed in https://github.com/bear-rsg/easybuild-easyblocks/pull/17, this custom `sanity_check_paths` should be stripped out

Comment (lexming) [reply easybuild/easyconfigs/m/Mesa/Mesa-20.0.2-GCCcore-9.3.0.eb]: The sanity checks in your PR will fail with this version of Mesa, see my comment at https://github.com/bear-rsg/easybuild-easyblocks/pull/17#discussion_r396122428

Comment (boegel) [reply easybuild/easyconfigs/m/Mesa/Mesa-20.0.2-GCCcore-9.3.0.eb]: fixed now, see https://github.com/easybuilders/easybuild-easyblocks/pull/1892

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/2ad7cfdf917bf996f8b573b8e86ad3bb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3117.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/148ed89b77234f7c1f2d803a350dae8f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2501.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/7395cbe00c877e5549218ca77e43c37e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/ad6c4d3a8a702615d27d371e8847250c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3117.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/37d8df0132b27cb54ff1b231f59d73ed for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2501.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/17e17c5bbe0a56be3ae5d640df09a1bc for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b94c0f84eabec08726b87c2f63f7817e for a full test report.

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**